src/Hintman/Core/Review.hs

```diff
@@ -0,0 +1,70 @@
+{- | Data types and instances for PR review.
+-}
+
+module Hintman.Core.Review
+       ( PullRequestReview (..)
+       , ReviewEvent (..)
+       , Comment (..)
+       , ReviewResponse (..)
+       ) where
+
+import Data.Aeson (object, withObject, (.:), (.=))
+
+
+data PullRequestReview = PullRequestReview
+    { prrBody     :: !Text
+    , prrEvent    :: !ReviewEvent
+    , prrComments :: ![Comment]
+    }
+
+instance ToJSON PullRequestReview where
+    toJSON PullRequestReview{..} = object
+        [ "body"     .= prrBody
+        , "event"    .= prrEvent
+        , "comments" .= prrComments
+        ]
+
+data ReviewEvent
+    = Approved
+    | ChangesRequested
+    | Commented
+
+-- Question: should I name constructors in CAPS in the same way?
+instance ToJSON ReviewEvent where
+    toJSON = \case
+        Approved         -> "APPROVE"
+        ChangesRequested -> "REQUEST_CHANGES"
+        Commented        -> "COMMENT"
+
+data Comment = Comment
+    { -- | Path to file
+      commentPath     :: !Text
+
+      {- | The position in the diff where you want to add a review comment. Note
+      this value is not the same as the line number in the file. For help
+      finding the position value, read the note below.
+      -}
+    , commentPosition :: !Int
+
+      -- | Text of the review comment
+    , commentBody     :: !Text
+    }
+
+instance ToJSON Comment where
+    toJSON Comment{..} = object
+        [ "path"     .= commentPath
+        , "position" .= commentPosition
+        , "body"     .= commentBody
+        ]
+
+-- response is big but for now we don't need much of it
+data ReviewResponse = ReviewResponse
+    { rrCommitId :: !Text
+    , rrHtmlUrl  :: !Text
```

Comment:
Hmm, do we need this data type? Looks not that informative as we would have all this info before...

Reply:
Well, every GitHub library returns something and we need to have a data type for it. I will try to use something simpler to introduce our own data type